setup.md

```diff
@@ -269,8 +269,124 @@ TODO: Explain one of the following:
 * What significance does this have to the user?
 * What's in it for the user? WIIFM (What's in it for me? (for the user))
 
-### To add readline support to Fennel
+The command-line REPL that comes with the `fennel` works out of the box, but
+the built-in line-reader is very limited in user experience. Adding
+[GNU Readline](https://tiswww.case.edu/php/chet/readline/rltop.html) support
+enables such user-friendly features as
+- Tab completion on the REPL that can complete on all locals, macros, and special forms
+- A rolling history buffer, which can be navigated, searched (`ctrl+r`), and optionally
+persisted to disk so you can search input from previous REPL sessions.
+- emacs (default) or vi emulation via readline's custom support for better line
+navigation
+- Optional use of additional readline features in `~/.inputrc`, such as blinking
+on matched parentheses or color color output
 
-1. TODO
-2. TODO
-3. TODO
+### Requirements for readline support
+
+All you need to enable readline support is:
+- GNU Readline installed on your system (you may already have it!)
+- [readline.lua](https://pjb.com.au/comp/lua/readline.html) lua bindings to libreadline
+
+The Fennel CLI REPL will automatically load and use the readline bindings when
+it can resolve the readline module, so that's all you need to get started.
+
+### Installing readline.lua
+For the official support on getting readline.lua, see the
+[official docs](https://pjb.com.au/comp/lua/readline.html#installation).
+
+The easiest way to get readline.lua is to install it with Luarocks, which
+will fetch the package and automatically compile the native bindings for you.
+
+#### Installing readline with LuaRocks
+
+```bash
+# to install globally on the system (requires admin privileges or sudo)
+$ luarocks install readline
+
+# to install to the user's local tree
+$ luarocks install --local readline
+
+# install for Lua 5.1 (including LuaJIT)
+$ luarocks install --lua-version=5.1 readline
+```
+
+Because the readline Lua module contains native bindings to libreadline, be sure
+it's installed for the Lua version you intend to use.
+
+If you've installed with the `--local` flag, you may need to ensure your `package.path`
+and `package.cpath` contain its location so it can be resolved by `require`.
+
+**TODO: Add and link to section on configuring environment to detect local rocks**
+
+#### Manual installation
```

Comment:
TBH I think for a manual installation we should just refer to the upstream docs. I don't think it belongs in a general setup guide.